### ISSUE TITLE
Breaks `import attrs`

### ISSUE THREAD

**Issue by ArthurGarnier** [mtkennerly/poetry-dynamic-versioning#85]
HI,

When installed in a venv with poetry-core it seems to breaks importing "attrs" module :

```
╰─$ python                  
Python 3.6.8 (default, Jan 14 2022, 11:04:20) 
[GCC 8.5.0 20210514 (Red Hat 8.5.0-7)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import attrs
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ldap/garnieat2/venv-perso/lib64/python3.6/site-packages/poetry_dynamic_versioning/patch.py", line 106, in alt_import
    module = _state.original_import_func(name, globals, locals, fromlist, level)
  File "/home/ldap/garnieat2/venv-perso/lib64/python3.6/site-packages/attrs/__init__.py", line 3, in <module>
    from attr import (
ImportError: cannot import name 'cmp_using'
```

Only happens if there is a pyproject.toml file in current dir or an parent one. 

Repoducting : 

```
python3 -m venv test
source test/bin/activate
pip install poetry-core poetry-dynamic-versioning attrs

cat << EOF > pyproject.toml
[tool.poetry]
name = "test"
description = ""
version = "0.0.0"


[tool.poetry-dynamic-versioning]
enable = true
EOF

python3 -m attrs
```

**Comment by mtkennerly** [comment]
I don't know exactly why, but import sequence issues like this tend to happen with `--user` installs, as opposed to non-user installs which typically work better. I wonder if venv installs are susceptible to the same issue, but I still don't understand the root cause, so it's hard to say.

While not a solution to your problem, would you be able to switch to a global/non-user install of Poetry and the plugin instead? That's the most stable environment for the plugin to avoid import sequence issues.

**Comment by mtkennerly** [comment]
To reconcile this and other recurring issues, I've just released [v0.18.0](https://github.com/mtkennerly/poetry-dynamic-versioning/releases/tag/v0.18.0), which replaces the import hack with a thin wrapper around Poetry Core for use in the `build-system` section of pyproject.toml.

If you still have any issues with the new version, please let me know, but it should eliminate the kind of conflict that you ran into.
